cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -0,0 +1,299 @@
+      shellQuote: false
+    - valueFrom: $(inputs.file)
+      shellQuote: false
+    - valueFrom: '|'
+      shellQuote: false
+    - valueFrom: wc
+      shellQuote: false
+    - valueFrom: -l
+      shellQuote: false
+    stdout: cgrep-stdOut.txt
+    id: file:///Users/danb/wdl4s/r.cwl#cgrep/09f8bcac-a91a-49d5-afb6-2f1b1294e875
+- id: file:///Users/danb/wdl4s/r.cwl#wc
+  in:
+  - id: file:///Users/danb/wdl4s/r.cwl#wc/file
+    source: file:///Users/danb/wdl4s/r.cwl#ps/ps-stdOut
+  out:
+  - id: file:///Users/danb/wdl4s/r.cwl#wc/wc-count
+  run:
+    inputs:
+    - id: file:///Users/danb/wdl4s/r.cwl#wc/45d98851-7bfe-473e-ab24-aac922553f3e/file
+      type: File
+    outputs:
+    - id: file:///Users/danb/wdl4s/r.cwl#wc/45d98851-7bfe-473e-ab24-aac922553f3e/wc-count
+      outputBinding:
+        glob: wc-stdOut.txt
+        loadContents: true
+        outputEval: $(self[0].contents.toInt)
+      type: int
+    class: CommandLineTool
+    requirements:
+    - class: ShellCommandRequirement
+    - class: InlineJavascriptRequirement
+    arguments:
+    - valueFrom: cat
+      shellQuote: false
+    - valueFrom: $(inputs.file)
+      shellQuote: false
+    - valueFrom: '|'
+      shellQuote: false
+    - valueFrom: wc
+      shellQuote: false
+    - valueFrom: -l
+      shellQuote: false
+    stdout: wc-stdOut.txt
+    id: file:///Users/danb/wdl4s/r.cwl#wc/45d98851-7bfe-473e-ab24-aac922553f3e
+id: file:///Users/danb/wdl4s/r.cwl
+"""
+
+    val workflow = CwlCodecs.decodeCwl(threeStep) match {
+      case Valid((wf: Workflow, _)) => wf
+      case o => fail(s"didn't parse a workflow! $o")
+    }
+    val wf = workflow.womExecutable(Map.empty) match {
+      case Valid(Executable(wf: WorkflowDefinition)) => wf
+      case o => fail(s"invalid executable $o")
+    }
+
+    val nodes = wf.innerGraph.nodes
+
+    nodes collect { case gin: GraphInputNode => gin.name } should be(Set("file:///Users/danb/wdl4s/r.cwl#pattern"))
+    nodes collect { case gon: GraphOutputNode => gon.name } should be(Set(
+      "file:///Users/danb/wdl4s/r.cwl#cgrep-count",
+      "file:///Users/danb/wdl4s/r.cwl#wc-count"
+    ))
+    nodes collect { case cn: CallNode => cn.name } should be(
+      Set(
+        "file:///Users/danb/wdl4s/r.cwl#ps",
+        "file:///Users/danb/wdl4s/r.cwl#cgrep",
+        "file:///Users/danb/wdl4s/r.cwl#wc"))
+
+    val ps = nodes.collectFirst({ case ps: CallNode if ps.name == "file:///Users/danb/wdl4s/r.cwl#ps" => ps }).get
+    val cgrep = nodes.collectFirst({ case cgrep: CallNode if cgrep.name == "file:///Users/danb/wdl4s/r.cwl#cgrep" => cgrep }).get
+    val cgrepPatternInput = nodes.collectFirst({ case cgrepInput: GraphInputNode if cgrepInput.name == "file:///Users/danb/wdl4s/r.cwl#pattern" => cgrepInput }).get
+    val wc = nodes.collectFirst({ case wc: CallNode if wc.name == "file:///Users/danb/wdl4s/r.cwl#wc" => wc }).get
+
+    ps.upstream shouldBe empty
+    cgrep.upstream shouldBe Set(ps, cgrepPatternInput)
```

Comment:
Are these `shouldBe`s based on object equality or `case class` equality?

I presume it's `case class` equality... is there any way of instead checking object equality?